src/test/java/io/jenkins/pluginhealth/scoring/probes/ContributingGuidelinesProbeTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Jenkins Infra
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class ContributingGuidelinesProbeTest {
```

Comment:
should the probe be executed when there are code changes on the repository? 
You need to test for that.

Reply:
I think I have a gap in my understanding. It would be really helpful if you could clarify it.

**What I think is happening**: Every two hours, the program is fetching all the plugins from the update center, and running all the probes on them which includes cloning of the repositories and searching of files like `CONTRIBUTING.md` and then modifying the scores automatically.

By 'code changes in the repository' I think you mean whenever someone merges a new code with the main branch of the plugin.

**My Question**: The present tests (inside the unit test file) can verify that the probe is able to find the contributing guidelines, why do we need to test if the probe is executing whenever there are code changes in the repository if the complete process is happening every two hours?

Also, `Jenkinsfile` probe unit test also does not contain anything similar due to which I am very confused.

Reply:
The problem is not the probe is or isn't able to find the file, but its conditions to be executed. 
In the `Probe` class, there are methods which help the `ProbeEngine` to determine if the probe should or shouldn't be executed:
 - `requiresRelease()`
 - `isSourceCodeRelated()`

It seems that I didn't add documentation on those methods so it's my fault. 

But, the first one specified that only a release could change the outcome of a probe on a plugin. So, if there is not new release since the last time a probe was executed on a plugin, there is no need to run the probe (the result would be the same as last time).

The second method specified that the probe requires code modifications to have a different outcome from the last execution.

By default, both methods are `false`. In your case, `isSourceCodeRelated` should be `true`, `requiresRelease` should be `false`.

I'll add documentation on those methods.

Reply:
> But, the first one specified that only a release could change the outcome of a probe on a plugin. So, if there is not new release since the last time a probe was executed on a plugin, there is no need to run the probe (the result would be the same as last time).
> 
Just to be clear my confusion, if `requiresRelease` is `false`, it means the probe would run every time regardless if there is a new release or not.

Reply:
> Just to clear my confusion, if requiresRelease is false, it means the probe would run every time regardless if there is a new release or not as it is not limited by waiting for a new release.

correct.

> If my statement above is correct, then making isSourceCodeRelated true won't have any effect (is redundant) as the probe would be executed every time on code change based on my above statement.

So, ok, both condition could be redundant, as, when you have a release, you have code change that is true. But you can have source code change without release. That is why I created those two methods.